Extensions/flashlight/Flashlight.json

```diff
@@ -0,0 +1,115 @@
+  "name": "FlashLight",
+  "previewIconUrl": "https://resources.gdevelop-app.com/assets/Icons/flashlight.svg",
+  "shortDescription": "This plugin allows you to switch the flashlight / torch of the device on and off.",
+  "version": "1.0.0",
+  "tags": [
+    "flashlight",
+    "android",
+    "light"
+  ],
+  "authorIds": [
+    "jy7FXnGX0ZZcWfrAI9YuQaeIphi1"
+  ],
+  "dependencies": [
+    {
+      "exportName": "cordova-plugin-flashlight",
+      "name": "Flash light",
+      "type": "cordova",
+      "version": "3.2.0"
+    }
+  ],
+  "eventsFunctions": [
+    {
+      "description": "switch the flashlight / torch of the device off ",
+      "fullName": "Switch Off flashlight / torch",
+      "functionType": "Action",
+      "name": "switchOff",
+      "private": false,
+      "sentence": "switch the flashlight / torch of the device off",
+      "events": [
+        {
+          "disabled": false,
+          "folded": false,
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "window.plugins.flashlight.available(function(isAvailable) {\n  if (isAvailable) {\n \n   \n     window.plugins.flashlight.switchOn(\n      function() {}, \n      function() {}, \n      {intensity: 0.3} \n    );\n \n  } else {\n    alert(\"Flashlight not found!\");\n  }\n});\n",
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": false
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "description": "switch the flashlight / torch of the device on ",
+      "fullName": "Switch On flashlight / torch",
+      "functionType": "Action",
+      "name": "switchOn",
+      "private": false,
+      "sentence": "switch the flashlight / torch of the device on ",
+      "events": [
+        {
+          "disabled": false,
+          "folded": false,
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "window.plugins.flashlight.available(function(isAvailable) {\n  if (isAvailable) {\n \n    window.plugins.flashlight.switchOff();\n \n  } else {\n    alert(\"Flashlight not found!\");\n  }\n});\n",
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": false
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "description": "you'll want to make sure the torch is switched off when the app is exited via the backbutton. Otherwise, the camera may be locked so it can't be used by other apps",
+      "fullName": "Switch off on app exit",
+      "functionType": "Action",
+      "name": "_onExit",
```

Comment:
`_onExit`
It is not the responsibility of the application to ensure that the lamp is switched off, it is the responsibility of the user.
Just remove this function.

Reply:
Or if you want to keep that action, maybe you can move it in a lifecycle function **OnSceneUnloading**.
Test it to be sure it working when you exit the app!